src/main/java/ui/diet/dietmanager/DietManagerUi.java

```diff
@@ -29,27 +31,28 @@ public static void printBye() {
      *         else returns original string
      * @throws DateTimeParseException if the date string input is not a valid date
      */
-    public String extractDate(String input) throws DateTimeParseException, IllegalStateException {
+    public String extractDate(String input) throws DateTimeParseException, IllegalStateException,
+            InvalidDateFormatException {
         try {
             String dateString = input.split("/t")[0].split(" ", 2)[1].trim();
-            Pattern pattern = Pattern.compile("\\d{4}-\\d{2}-\\d{2}");
-            Matcher matcher = pattern.matcher(dateString);
-            boolean isValidDate = matcher.find();
-            String match = matcher.group();
-            return LocalDate.parse(match).format(DateTimeFormatter.ofPattern("MM-dd-yyyy"));
+            //Pattern pattern = Pattern.compile("\\d{4}-\\d{2}-\\d{2}");
+            //Matcher matcher = pattern.matcher(dateString);
+            //String match = matcher.group();
+            return DateParser.parseDate(dateString).format(DateTimeFormatter.ofPattern("dd-MM-yyyy"));
+            //return LocalDate.parse(match).format(DateTimeFormatter.ofPattern("dd-MM-yyyy"));
```

Comment:
Should we delete those commented out lines?

Reply:
will do